app/src/test/java/org/simple/clinic/monthlyscreeningreports/form/QuestionnaireEntryUpdateTest.kt

```diff
@@ -45,6 +45,30 @@ class QuestionnaireEntryUpdateTest {
         ))
   }
 
+  @Test
+  fun `when questionnaire response is fetched, then update the model`() {
+    val questionnaireResponse = TestData.questionnaireResponse()
+
+    spec
+        .given(defaultModel)
+        .whenEvent(QuestionnaireResponseFetched(questionnaireResponse))
+        .then(assertThatNext(
+            hasModel(defaultModel.responseLoaded(questionnaireResponse)),
+            hasNoEffects()
+        ))
+  }
+
+  @Test
+  fun `when questionnaire response is saved, then go to complete screen`() {
```

Comment:
Is there a test like `when a questionnaire response is saved, submitted is set to "true", and the input values are saved inside "content"`? Should add integration tests if we haven't.

Reply:
Yes, it should be added